dlc-manager/src/sub_channel_manager.rs

```diff
@@ -316,6 +316,11 @@ where
                     SubChannelState::OffChainClosed => {
                         s.is_offer = true;
                         s.update_idx -= 1;
+
+                        // We do this to give older channels a chance to upgrade to the new version which
```

Comment:
Honestly I'm not a big fan of this. It feels hacky and error prone. I'd like to have a way to handle upgrading at de-serialization time. Not sure what the best way is to be honest, but for example LDK uses the `ReadableArgs` trait for when deserialization requires external data.

Reply:
Yeah, it was supposed to be hacky in this case.

We don't have to merge this here anyway. I will just update the PR and we can just keep depending on this patch in 10101.

Reply:
If you think it's time to take backward compatibility issues seriously, it's probably worth taking the time to find a proper solution (though it's also hard to say how many more breaking changes will be required to reach a stable implementation).

Reply:
I agree with you on needing to figure out how to do this properly as soon as possible. But _we_ can probably live with this quick fix for now.